closure/compiler/closure_js_library.bzl

```diff
@@ -83,15 +85,16 @@ def _impl(ctx):
       execution_requirements={"supports-workers": "1"},
       progress_message="Checking %d JS files in %s" % (
           len(ctx.files.srcs) + len(ctx.files.externs), ctx.label))
-  return struct(files=set(ctx.files.srcs, order="compile"),
+  return struct(files=set(order="compile"),
                 js_language=determine_js_language(ctx),
                 js_exports=ctx.attr.exports,
                 js_provided=ctx.outputs.provided,
                 required_css_labels=collect_required_css_labels(ctx),
                 transitive_js_srcs=srcs,
                 transitive_js_externs=externs,
-                runfiles=ctx.runfiles(files=[ctx.outputs.provided,
-                                             ctx.outputs.stderr]))
+                transitive_data=tdata + ctx.files.data,
+                runfiles=ctx.runfiles(files=ctx.files.srcs + ctx.files.data,
```

Comment:
Shouldn't you be able to just use collect_default & co, and then wouldn't have to explicitly pass the transitive part?